experimental/saiexperimentaldashdevice.h

```diff
@@ -0,0 +1,134 @@
+/**
+ * Copyright (c) 2024 Microsoft Open Technologies, Inc.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License"); you may
+ *    not use this file except in compliance with the License. You may obtain
+ *    a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    THIS CODE IS PROVIDED ON AN *AS IS* BASIS, WITHOUT WARRANTIES OR
+ *    CONDITIONS OF ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT
+ *    LIMITATION ANY IMPLIED WARRANTIES OR CONDITIONS OF TITLE, FITNESS
+ *    FOR A PARTICULAR PURPOSE, MERCHANTABILITY OR NON-INFRINGEMENT.
+ *
+ *    See the Apache Version 2.0 License for specific language governing
+ *    permissions and limitations under the License.
+ *
+ *    Microsoft would like to thank the following companies for their review and
+ *    assistance with these files: Intel Corporation, Mellanox Technologies Ltd,
+ *    Dell Products, L.P., Facebook, Inc., Marvell International Ltd.
+ *
+ * @file    saiexperimentaldashdevice.h
+ *
+ * @brief   This module defines SAI P4 extension  interface
+ */
+
+#if !defined (__SAIEXPERIMENTALDASHDEVICE_H_)
+#define __SAIEXPERIMENTALDASHDEVICE_H_
+
+#include <saitypes.h>
+
+/**
+ * @defgroup SAIEXPERIMENTALDASH_DEVICE SAI - Extension specific API definitions
+ *
+ * @{
+ */
+
+/**
+ * @brief Attribute ID for DASH device
+ */
+typedef enum _sai_dash_device_attr_t
+{
+    /**
+     * @brief Start of attributes
+     */
+    SAI_DASH_DEVICE_ATTR_START,
+
+    /**
+     * @brief Any non-VNET VNI or GRE keys that need to be allowed
+     *
+     * @par PrivateLink GRE key is an example
+     *
+     * @type sai_vni_range_list_t
+     * @flags CREATE_ONLY
+     * @default empty
+     */
+    SAI_DASH_DEVICE_ATTR_TRUSTED_VNI_LIST = SAI_DASH_DEVICE_ATTR_START,
```

Comment:
we already have the VNI APIs in DASH, so we should use that to move forward without introducing the new one.

if any other things required on DPU level, they could go to switch extensions.